src/callbacks/keyring.h

```diff
@@ -273,8 +273,9 @@ namespace zypp
         // gpg key info
         dumpKeyInfo( s << std::endl, key_r, context_r )  << std::endl;
 
-        // if --gpg-auto-import-keys or --no-gpg-checks print info and don't ask
-        if (_gopts.gpg_auto_import_keys)
+        // if --gpg-auto-import-keys, --no-gpg-check, or --gpg-auto-import-key-id matches key, print info and don't ask.
+        if (_gopts.gpg_auto_import_keys
+            || (!_gopts.gpg_auto_import_key_id.empty() && _gopts.gpg_auto_import_key_id == key_r.id()))
```

Comment:
`_gopts.gpg_auto_import_key_id == key_r.id()` IMO is to strict.

The argument should be ID or FPR (recommended 16byte/40byte). As convenience a short-ID (8byte insecure) could be accepted. This would be `key_r.providesKey( _gopts.gpg_auto_import_key_id )`.

At least the manpage should state that the short-ID is considered to be insecure. Optionally the argparser could print a note/warning if a `! PublicKeyData::isSafeKeyId( _gopts.gpg_auto_import_key_id )` is passed.

Reply:
Change made in commit 180266c